src/NumericalAlgorithms/Interpolation/ZeroCrossingPredictor.cpp

```diff
@@ -0,0 +1,22 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Interpolation/ZeroCrossingPredictor.hpp"
+#include "NumericalAlgorithms/Interpolation/LinearLeastSquares.hpp"
+
+#include "DataStructures/DataVector.hpp"
+
+#include <vector>
+namespace intrp {
+
+double ZeroCrossingPredictor(const std::vector<double>& x_values,
+                             const std::vector<double>& y_values) {
+  intrp::LinearLeastSquares<1> predictor(x_values.size());
+  const auto coefficients = predictor.fit_coefficients(x_values, y_values);
+  double b = coefficients[0];
+  double m = coefficients[1];
+  double crossing = -1.0 * b / m;
+  return crossing;
```

Comment:
just combine the last four lines into `return -coefficients[0]/coefficients[1];`